gateway/mw_request_size_limit.go

```diff
@@ -64,6 +68,18 @@ func (t *RequestSizeLimitMiddleware) ProcessRequest(w http.ResponseWriter, r *ht
 
 	vInfo, _ := t.Spec.Version(r)
 
+	globalConfigMaxRequestBodySize := t.Spec.GlobalConfig.HttpServerOptions.MaxRequestBodySize
```

Comment:
No need to do this check 3 times. Just do smth like:
```
requestLimit := t.Spec.GlobalConfig.HttpServerOptions.MaxRequestBodySize
if vInfo.GlobalSizeLimit >0 && if vInfo.GlobalSizeLimit < requestLimit {
  requestLimit = vInfo.GlobalSizeLimit 
}

// If there's a potential match, try to match
found, meta := t.Spec.CheckSpecMatchesStatus(r, versionPaths, RequestSizeLimit)
if found {
  logger.Debug("Request size limit matched for this URL, checking...")
  rmeta := meta.(*apidef.RequestSizeMeta)
  if rmeta.SizeLimit > 0 && rmeta.SizeLimit  <  requestLimit {
     requestLimit = rmeta.SizeLimit
  }
}

if requestLimit == 0 {
 return nil, http.StatusOk
}

return t.checkRequestLimit(r, requestLimit)
```

Reply:
Or even better write small helper which returns request size with this logic, but also used in `EnabledForSpec`